feat(python): Document default span attrs in Celery, update for 3.0 (#14365)
getsentry/sentry-docs: docs/platforms/python/integrations/celery/index__v3.x.mdx

---
title: Celery
description: "Learn about using Sentry with Celery."
---

The Celery integration adds support for the [Celery Task Queue System](https://docs.celeryq.dev/).

## Install

Install `sentry-sdk` from PyPI with the `celery` extra:

```bash {tabTitle:pip}
pip install "sentry-sdk[celery]"
```
```bash {tabTitle:uv}
uv add "sentry-sdk[celery]"
```

## Configure

If you have the `celery` package in your dependencies, the Celery integration will be enabled automatically when you initialize the Sentry SDK.

<Alert>
  Make sure that the call to `sentry_sdk.init()` is loaded on worker startup and

>   not only in the module where your tasks are defined. Otherwise, the

Reviewer: ```suggestion
  not just in the module that defines your tasks. Otherwise, the
```

  - your Celery tasks will start a new trace and won't be connected to the trace in the calling function.

  The default is `True`.

  See [Distributed Traces](#distributed-traces) below to learn how to get more fine grained control over distributed tracing in Celery tasks.

- `monitor_beat_tasks`:

  Turn auto-instrumentation on or off for Celery Beat tasks using Sentry Crons.

  See <PlatformLink to="/crons/#celery-beat-auto-discovery">Celery Beat Auto Discovery</PlatformLink> to learn more.

  The default is `False`.

- `exclude_beat_tasks`:

  A list of Celery Beat tasks that should be excluded from auto-instrumentation using Sentry Crons. Only applied if `monitor_beat_tasks` is set to `True`.

  The list can contain strings with the names of tasks in the Celery Beat schedule to be excluded. It can also include regular expressions to match multiple tasks. For example, if you include `"payment-check-.*"` every task starting with `payment-check-` will be excluded from auto-instrumentation.

  See <PlatformLink to="/crons/#celery-beat-auto-discovery">Celery Beat Auto Discovery</PlatformLink> to learn more.

  The default is `None`.

## Behavior

### Distributed Traces

> Distributed tracing extends the trace from the code that's running your Celery task so that it includes the code that initiated the task.

Reviewer: ```suggestion
Distributed tracing connects the trace of your Celery task to the trace of the code that started the task, giving you a complete view of the entire workflow.
```

        ),
    ],
)

# This will propagate the trace:
my_task_a.delay("some parameter")

# This will propagate the trace:
my_task_b.apply_async(
    args=("some_parameter", )
)

# This will NOT propagate the trace. The task will start its own trace:
my_task_b.apply_async(
    args=("some_parameter", ),
    headers={"sentry-propagate-traces": False},
)

# Note: overriding the tracing behaviour using `task_x.delay()` is not possible.
```

### Default Span Attributes

A set of predefined span attributes will be attached to Celery transactions by default. These can also be used for sampling since they will also be accessible via the `sampling_context` dictionary in the [`traces_sampler`](/platforms/python/configuration/options/#traces_sampler).

  | Span Attribute              | Description                                           |
  | --------------------------- | ----------------------------------------------------- |
  | `celery.job.args.{index}`   | Positional arguments provided to the task, serialized |
  | `celery.job.kwargs.{kwarg}` | Keyword arguments provided to the task, serialized    |
  | `celery.job.task`           | Task name                                         |

> These attributes will also be sent to Sentry. If you don't want that, you can filter them out using a custom [`before_send`](/platforms/python/configuration/options/#before_send) function.

Author: This is new, rest is copied from the base page

> These attributes will also be sent to Sentry. If you don't want that, you can filter them out using a custom [`before_send`](/platforms/python/configuration/options/#before_send) function.

Author: This is new, rest is copied from the base page

## Supported Versions

> - Celery: 4.4.7+

Author: Updated version here
